src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreResult.cs

```diff
@@ -56,6 +58,12 @@ public class RestoreResult : IRestoreResult
         /// </summary>
         public TimeSpan ElapsedTime { get; }
 
+        /// <summary>
+        /// The log messages raised during this restore operation
+        /// </summary>
+        /// <remarks>The messages here are usually sources from the <see cref="LockFile"/> in full restores or <see cref="CacheFile"/> for no-op restores.</remarks>
+        public virtual IList<IAssetsLogMessage> LogMessages { get; internal set; }
```

Comment:
Tests to make sure `LogMessages` has correct value?

Reply:
Good point, added tests.